File: katas/FizzBuzz/solutions/marc/RubyFizzBuzz/FizzBuzz.rb
class FizzBuzzCalculator
    def calc
        for value in 1..100 do
            puts naivImplementation(value)
        end
    end

    def naivImplementation(number)
        if number % 15 == 0      
            "FizzBuzz"
        elsif number % 5 == 0
            "Buzz"
        elsif number % 3 == 0
            "Fizz"
        else
            number.to_s
        end
    end

    def simpleImplementation(number)
        result = ""
        if number % 3 == 0
            result = "Fizz"
        end
        if number % 5 == 0
            result = result + "Buzz"
        end
        if result.empty?
            result = number.to_s
        end
        result
    end
end

File: katas/FizzBuzz/solutions/kepp_fizzbuzz5minds/js/fizzbuzz.js
exports.getValueFor = (number) => {
  if (number % (3 * 5) === 0) return "FizzBuzz";
  if (number % 3 === 0) return "Fizz";
  if (number % 5 === 0) return "Buzz";
  return number.toString();
}

exports.getValueSpecial = (number) => {
  const result = number.toString();
  if (result.includes("5") && result.includes("3")) return "FizzBuzz";
  if (result.includes("3")) return "Fizz";
  if (result.includes("5")) return "Buzz";
  return this.getValueFor(parseInt(result));
}
